[📖 Docs]: Improve TypeScript documentation for extending `expect`

### Pre-check

- [X] I know I can edit the docs but prefer to file this issue

### Describe the improvement

Unclear documentation

### Description of the improvement / report

Currently, the [docs](https://webdriver.io/docs/api/expect-webdriverio/#adding-your-own-matchers) for extending WDIO's `expect` with a custom matcher only use JavaScript as an example.

While following these docs does work, TypeScript users will run into compiler errors. This is because the normal method for extending the `expect` type does not seem to have any effect on the `expect` from `expect.webdriverio`:

```ts
declare module 'expect' {
  interface AsymmetricMatchers {
    toBeWithinRange(floor: number, ceiling: number): void;
  }
  interface Matchers<R> {
    toBeWithinRange(floor: number, ceiling: number): R;
  }
}
```
```
error TS2339: Property 'toBeWithinRange' does not exist on type 'Matchers<void, number>'
```

It would be nice if the docs included a snippet for TypeScript users demonstrating how to properly extend the type for `expect` when extending it with custom matchers.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/examples/wdio/vite-vue-example/package.json
+++ b/examples/wdio/vite-vue-example/package.json
@@ -17,7 +17,7 @@
     "@testing-library/vue": "^6.6.1",
     "@types/mocha": "^10.0.1",
     "@vitejs/plugin-vue": "^4.0.0",
-    "expect-webdriverio": "^4.1.2",
+    "expect-webdriverio": "^4.8.0",
     "mocha": "^10.2.0",
     "typescript": "^4.9.5",
     "vite": "^4.1.5",

--- a/packages/wdio-browser-runner/package.json
+++ b/packages/wdio-browser-runner/package.json
@@ -44,7 +44,7 @@
     "@wdio/utils": "8.27.0",
     "ast-types": "^0.14.2",
     "deepmerge-ts": "^5.0.0",
-    "expect-webdriverio": "^4.6.1",
+    "expect-webdriverio": "^4.8.0",
     "fast-safe-stringify": "^2.1.1",
     "get-port": "^7.0.0",
     "import-meta-resolve": "^4.0.0",

--- a/packages/wdio-globals/package.json
+++ b/packages/wdio-globals/package.json
@@ -42,7 +42,7 @@
     "access": "public"
   },
   "optionalDependencies": {
-    "expect-webdriverio": "^4.6.1",
+    "expect-webdriverio": "^4.8.0",
     "webdriverio": "8.27.0"
   }
 }

--- a/packages/wdio-jasmine-framework/package.json
+++ b/packages/wdio-jasmine-framework/package.json
@@ -37,7 +37,7 @@
     "@wdio/logger": "8.24.12",
     "@wdio/types": "8.27.0",
     "@wdio/utils": "8.27.0",
-    "expect-webdriverio": "^4.6.1",
+    "expect-webdriverio": "^4.8.0",
     "jasmine": "^5.0.0"
   },
   "devDependencies": {

--- a/packages/wdio-runner/package.json
+++ b/packages/wdio-runner/package.json
@@ -36,7 +36,7 @@
     "@wdio/types": "8.27.0",
     "@wdio/utils": "8.27.0",
     "deepmerge-ts": "^5.0.0",
-    "expect-webdriverio": "^4.6.1",
+    "expect-webdriverio": "^4.8.0",
     "gaze": "^1.1.2",
     "webdriver": "8.27.0",
     "webdriverio": "8.27.0"

--- a/website/_sidebars.json
+++ b/website/_sidebars.json
@@ -48,6 +48,7 @@
         "emulation",
         "mocksandspies",
         "customcommands",
+        "custommatchers",
         "pageobjects",
         "debugging",
         "repl",

--- a/website/docs/CustomMatchers.md
+++ b/website/docs/CustomMatchers.md
@@ -1,0 +1,48 @@
+---
+id: custommatchers
+title: Custom Matchers
+---
+
+WebdriverIO uses a Jest style [`expect`](https://webdriver.io/docs/api/expect-webdriverio) assertion library that comes with special features and custom matchers specific for running web and mobile tests. While the library of matchers is big, it certainly doesn't fit all possible situations. Therefore it is possible to extend the existing matchers with custom ones defined by you.
+
+:::warning
+
+While there is currently no difference in how matchers are defined that are specific to the [`browser`](/docs/api/browser) object or an [element](/docs/api/element) instance, this certainly might change in the future. Keep an eye on [`webdriverio/expect-webdriverio#1408`](https://github.com/webdriverio/expect-webdriverio/issues/1408) for further information on this development.
+
+:::
+
+## Custom Browser Matchers
+
+To register a custom browser matcher, call `extend` on the `expect` object either in your spec file directly or as part of the e.g. `before` hook in your `wdio.conf.js`:
+
+```js reference useHTTPS
+https://github.com/webdriverio/example-recipes/blob/e719632df8f241f923c8d9301aab6bccee5cb109/customMatchers/example.ts#L3-L18
+```
+
+As shown in the example the matcher function takes the expected object, e.g. the browser or element object, as the first parameter and the expected value as the second. You can then use the matcher as follows:
+
+```js reference useHTTPS
+https://github.com/webdriverio/example-recipes/blob/e719632df8f241f923c8d9301aab6bccee5cb109/customMatchers/example.ts#L50-L52
+```
+
+## Custom Element Matchers
+
+Similar to custom browser matchers, element matchers don't differ. Here is an example of how to create a custom matcher to assert the aria-label of an element:
+
+```js reference useHTTPS
+https://github.com/webdriverio/example-recipes/blob/e719632df8f241f923c8d9301aab6bccee5cb109/customMatchers/example.ts#L20-L38
+```
+
+This allows you to call the assertion as follows:
+
+```js reference useHTTPS
+https://github.com/webdriverio/example-recipes/blob/e719632df8f241f923c8d9301aab6bccee5cb109/customMatchers/example.ts#L54-L57
+```
+
+## TypeScript Support
+
+If you are using TypeScript, one more step is required to ensure the type safety of your custom matchers. By extending the `Matcher` interface with your custom matchers, all type issues vanish:
+
+```js reference useHTTPS
+https://github.com/webdriverio/example-recipes/blob/e719632df8f241f923c8d9301aab6bccee5cb109/customMatchers/example.ts#L40-L47
+```

